### PR TITLE
[FIX] disable_invalid_filters: <module 'time' (built-in)> can not be used in evaluation contexts

### PR DESCRIPTION
**Pull request by royleviindoo**
This error will save archive all filters
=> Prefer providing only the items necessary for your intended use as attributes of `odoo.tools.safe_eval`

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This has been working perfectly during several versions. Is it a problem in v14? Why? If so, you must restrict it to such version, not on the rest.

**Comment by royleviindoo** [comment]
> This has been working perfectly during several versions. Is it a problem in v14? Why? If so, you must restrict it to such version, not on the rest.

Yes, it is a problem in v14 (all filters is archived)
How to restrict it to such version?

**Comment by pedrobaeza** [comment]
But I don't get why. Having an extra argument shouldn't be the reason for that. I have migrated to v14 already some DBs and don't have this problem.

**Comment by royleviindoo** [comment]
> But I don't get why. Having an extra argument shouldn't be the reason for that. I have migrated to v14 already some DBs and don't have this problem.

Because if error occus, except clause is handled, so all filters is archived
You are checking all filters in db after migrated? You can check at menu Settings / Technical / User-defined Filters

**Comment by royleviindoo** [comment]
> But I don't get why. Having an extra argument shouldn't be the reason for that. I have migrated to v14 already some DBs and don't have this problem.
> 

The following code generates that message, however, I see no need to use `time` in the safe_eval

https://github.com/odoo/odoo/blob/14.0/odoo/tools/safe_eval.py#L366-L372

**Comment by royleviindoo** [comment]
@pedrobaeza I fixed to restrict it to version 14.0, please review, thank you

File: openupgradelib/openupgrade.py [hunk 1]
```diff
@@ -2239,13 +2239,17 @@ def format_message(f):
         columns = (
             getattr(model, '_columns', False) or getattr(model, '_fields')
         )
+
+        localdict = {'uid': env.uid}
```
Comment (MiquelRForgeFlow) [inline openupgradelib/openupgrade.py:2243]:
```suggestion
        globaldict = {'uid': env.uid}
```
Selection_245

Comment (royleviindoo) [reply openupgradelib/openupgrade.py]: @MiquelRForgeFlow I updated, thank you

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by royleviindoo** [comment]
@pedrobaeza please review this PR, thank you

**Comment by all-dm** [comment]
We're migrating a db from v12 to v14 and found this problem too. It's something new in v14 that I didn't see before. Because of this we're getting around 408 user-defined filters archived, so would be great when this pull request can be reviewed and merged.

**Review by pedrobaeza — APPROVED** [review]
Checked that no other replacement is found, so good to go

**Pull request merged**